macros/utils/aggregations/_min_or_max.sql

```diff
@@ -1,4 +1,8 @@
 {% macro _min_or_max(min_or_max, qualified_col) %}
+    {{ return(adapter.dispatch('_min_or_max','dbt_activity_schema')(min_or_max, qualified_col)) }}
+{%- endmacro -%}
+
+{% macro default___min_or_max(min_or_max, qualified_col) -%}
 
 {% set aggregation = "min" if min_or_max == "min" else "max" %}
```

Comment:
I think this assignment should be via a dispatched macro. And then we can call that aggregation with `{{ }}` in the relevant place.

Reply:
@tnightengale Apologies for the back and forth. I think I'm missing something here that would help me address this feedback. Three questions for you:

1. I follow how the `default` `_min_or_max` macro works- per the discussion in #26, this code uses built-in Snowflake functionality to achieve the same end rather than replicate the prepend approach. In my dbt project and the Snowflake version of the integration tests, I see that the `min_by/max_by` version produces the expected results. Could you explain further what you mean by `I don't think this will work as is?`?
2. For `So I think that logic needs to be replicated for Snowflake as well, as far as I understand.`,  are you saying that you would prefer a Snowflake version of the prepend logic rather than the native Snowflake approach used here or something else?
3. I'm not following the comment on the assignment via dispatch macro. Could you explain more about what you are looking for or point me to an example on which I can base the refactoring?

Appreciate the feedback to give me what I need to get this all squared away!